### DISCUSSION TITLE
OC 2.0 and Taxonomy Autoroute liquid patterns

### DISCUSSION THREAD

**Discussion by kevinasdzine** [OrchardCMS/OrchardCore#16832] [Q&A] [answered]
My OC 2.0 prototype is consistently producing incorrect Autoroute permalinks with taxonomies. It never includes the beginning of the URL pattern that is the taxonomy itself. For example, I have a taxonomy named "Location" and URL's should have "/location/..." at the beginning but they don't.

Also, I have this pattern for the TermType's Autoroute but it currently results in the term's own slug being used twice (e.g. _meeting-room-1/meeting-room-1_), not the parents:
`{{ Model.ContentItem | container | display_text | slugify }}/{{ Model.ContentItem | display_text | slugify }}`

Is there a current document that describes what Autoroute liquid patterns should be used for the Taxonomy type and the various TermTypes that are created for different taxonomies?

**Comment by Piedone** [comment]
Can you show your attempt, what's not working?

**Comment by kevinasdzine** [comment]
This is with `{{ Model.ContentItem | container | display_text | slugify }}/{{ Model.ContentItem | display_text | slugify }}` as the pattern for the TermType Autoroute which, as far as I know, should have the parent path and the child path (it's using the child path on both sides of the "/" character). The upper link should be the parent of the lower link

![image](https://github.com/user-attachments/assets/3232f3ef-6387-4f51-9c4c-420735d536b1)

**Reply by kevinasdzine** [reply]
Ooops, this is a problem with the Themed version of the taxonomy display, not the the default part view.

However, that raises a related question. When emitting a taxonomy term view in your own template, is there a shortcut for building the parent path for the current term?

**Reply by Piedone** [reply]
You can use this:

```html
<a display-for="@contentItem">...</a>
```

Be sure to reference `OrchardCore.Contents.TagHelpers` in the project.

**Reply by kevinasdzine** [reply, marked as answer]
Never mind, got that with the `GetInheritedTermsAsync` Orchard Helper. Problem solved.
